project/Assets/Obstacle/ObstacleManager.cs

```diff
@@ -20,16 +20,21 @@ void OnTriggerEnter2D(Collider2D intruderCollider)
     {
         GameObject intruder = intruderCollider.gameObject;
```

Comment:
What do you mean with intruder? maybe this needs a better name to define that's a object passing between the pipes